Software 0.29 documentation (#756)
astronomer/docs: software/release-notes.md

<!--- Version-specific -->

This document includes all release notes for Astronomer Software version 0.29.

> 0.29 is the latest stable version of Astronomer Software, while 0.28 remains  the latest LTS long-term support (LTS) version of Astronomer Software. To upgrade to 0.29, read [Upgrade Astronomer](upgrade-astronomer-stable.md). For more information about Software release channels, read [Release and Lifecycle Policies](release-lifecycle-policy.md). To read release notes specifically for the Astronomer CLI, see [Astronomer CLI Release Notes](cli-release-notes.md).

Reviewer: ```suggestion
0.29 is the latest stable version of Astronomer Software, while 0.28 remains the latest LTS long-term support (LTS) version of Astronomer Software. To upgrade to 0.29, read [Upgrade Astronomer](upgrade-astronomer-stable.md). For more information about Software release channels, read [Release and Lifecycle Policies](release-lifecycle-policy.md). To read release notes specifically for the Astro CLI, see [Astro CLI Release Notes](cli-release-notes.md).
```

We're committed to testing all Astronomer Software versions for scale, reliability and security on Amazon EKS, Google GKE and Azure AKS. If you have any questions or an issue to report, don't hesitate to [reach out to us](https://support.astronomer.io).

## v0.29.0

Release date: June 1, 2022

> ### Support for Astro Runtime Images on Software

Reviewer: ```suggestion
### Support for Astro Runtime Images
```

Implied I think

You can now use Astro Runtime images in your Software Deployments. Additionally, you can now select Runtime images when setting **Image Version** for a Deployment in the Software UI.

Functionally, Runtime images are similar to Certified images. They both include:

- Same-day support for Apache Airflow releases
- Extended support lifecycles
- Regularly backported bug and security fixes

Astronomer Runtime includes additional features which are not available in Astronomer Certified images, including:

> - The `astronomer-providers` package, which includes a set of deferrable operators build and maintained by Astronomer

Reviewer: ```suggestion
- The `astronomer-providers` package, which includes a set of operators that are built and maintained by Astronomer
```

- Airflow UI improvements, such as the showing your Deployment's Docker image tag in the footer of all UI pages

> - Future Runtime-exclusive features, such as new Airflow components and improvements to the DAG development experience

Reviewer: ```suggestion
- Features that are exclusive to Astro Runtime and coming soon, such as new Airflow components and improvements to the DAG development experience
```

> To upgrade a Deployment to Runtime, follow the steps in Upgrade Airflow, making sure to replace the Astronomer Certified image in your Dockerfile with an Astronomer Runtime version.

Reviewer: ```suggestion
To upgrade a Deployment to Runtime, follow the steps in [Upgrade Airflow](LINK?), making sure to replace the Astronomer Certified image in your Dockerfile with an Astro Runtime version.
```
